### ISSUE TITLE
add boto3 optional dependency?

### ISSUE THREAD

**Issue by croth1** [conda-forge/mlflow-feedstock#115]
### Comment:

MLflow has a variety of extra dependencies and the conda-forge ecosystem does not really support optional dependencies well. I wonder whether not depending on boto3 was a conscious decision or an oversight. I tripped over it today.

https://github.com/mlflow/mlflow/blob/38e1d481f427b94af8f471e1658b4971ca6391b5/setup.py#L136

**Comment by xhochy** [comment]
Extra requires are not needed for the core functionality. I would like to avoid having them as dependencies of the main output. You could add an additional output called mlflow-extras that would install all the extra dependencies.
